relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py

```diff
@@ -118,6 +119,43 @@ def dictToAsciiTable(data):
            break
     return outmsg
 
+
+# Use an authenticated proxy to access download.rudder.io, requests can only do basic auth, but we are not doing it here nonetheless
+# Got the code from a SO answer: https://stackoverflow.com/a/13520486/1939653
+class HTTPProxyDigestAuth(requests.auth.HTTPDigestAuth):
+    def handle_407(self, r):
+        """Takes the given response and tries digest-auth, if needed."""
+        num_407_calls = r.request.hooks['response'].count(self.handle_407)
+        s_auth = r.headers.get('Proxy-authenticate', '')
+        if 'digest' in s_auth.lower() and num_407_calls < 2:
+            self.chal = requests.auth.parse_dict_header(s_auth.replace('Digest ', ''))
+            # Consume content and release the original connection
+            # to allow our new request to reuse the same one.
+            r.content
+            r.raw.release_conn()
+            r.request.headers['Authorization'] = self.build_digest_header(r.request.method, r.request.url)
+            r.request.send(anyway=True)
+            _r = r.request.response
+            _r.history.append(r)
+            return _r
+        return r
+    def __call__(self, r):
+        if self.last_nonce:
+            r.headers['Proxy-Authorization'] = self.build_digest_header(r.method, r.url)
+        r.register_hook('response', self.handle_407)
+        return r
+
+def getRequest(url, stream):
+  if (PROXY_URL == ""):
+    return requests.get(completeUrl, auth=(USERNAME, PASSWORD), stream=stream)
+  else:
+    proxies = { "https": PROXY_URL }
```

Comment:
shouldn't we handle http too ?

Reply:
I think all request are made through https but maybe yes

espcially that it's authenticated requests with password that would leak the password